src/core/services/interimElement/interimElementProvider.js

```diff
@@ -0,0 +1,274 @@
+        angular.extend({}, defaultOptions, opts)
+      );
+    }
+
+    /**
+     * Helper to call $injector.invoke with a local of the factory name for
+     * this provider.
+     * If an $mdDialog is providing options for a dialog and tries to inject
+     * $mdDialog, a circular dependency error will happen.
+     * We get around that by manually injecting $mdDialog as a local.
+     */
+    function invokeFactory(factory, defaultVal) {
+      var locals = {};
+      locals[interimName] = publicService;
+      return $injector.invoke(factory || function() { return defaultVal; }, {}, locals);
+    }
+  }
+}
+
+function InterimElementService($$interimElementBase, $q) {
+  // This is needed, otherwise ng-annotator can't detect this function as a factory $get.
+  'ngInject';
+
+  return function() {
+    var API;
+    var SHOW_CANCELLED = false;
+    var showQueue = [];
+
+    return API = {
+      showInterim: _showInterim,
+      hideInterim: _hideInterim,
+      cancelInterim: _cancelInterim,
+      destroyInterim: _destroyInterim
+    };
+
+    function _showInterim(options) {
+      options = options || {};
+      // We inject the actual interim service into the options.
+      // This is needed for hideDelay etc. because it will call the service to hide the interim.
+      // This important, because otherwise the showQueue won't be cleared.
+      options._service = API;
+
+      var interimElement = new $$interimElementBase(options || {});
+      var hideExisting = !options.skipHide && showQueue.length ? API.hideInterim() : $q.when(true);
+
+      // Wait for the current existing interim to be hidden.
+      hideExisting.finally(function() {
+
+        showQueue.push(interimElement);
+
+        interimElement.show();
+
+      });
+
+      // Return a promise that will be resolved when the interim
+      // element is hidden or cancelled.
+      return interimElement.deferred.promise;
+    }
+
+    function _hideInterim(reason, options) {
+      if (!showQueue.length) return $q.when(reason);
+      options = options || {};
+
+      if (options.closeAll) {
+        var promise = $q.all(showQueue.reverse().map(closeElement));
+        showQueue = [];
+        return promise;
+      } else if (options.closeTo !== undefined) {
+        return $q.all(showQueue.splice(options.closeTo).map(closeElement));
+      }
+
+      var interim = showQueue.pop();
+      return closeElement(interim);
+
+      function closeElement(interim) {
+
+        interim.remove(reason, false, options);
+
+        return interim.deferred.promise;
+      }
+    }
+
+    function _cancelInterim(reason, options) {
+      var interim = showQueue.pop();
+      if (!interim) return $q.when(reason);
+
+      interim.remove(reason, true, options || {});
+
+      return interim.deferred.promise;
+    }
+
+    function _destroyInterim(target) {
+      var interim = !target ? showQueue.shift() : null;
+
+      var cntr = angular.element(target).length ? target.parentNode : null;
```

Comment:
You can also ditch the `angular.element` call here and just check if `target` is present.